js-pkg/sdk/src/main.ts

```diff
@@ -214,8 +219,8 @@ export class DocumentManager {
    *
    * @returns A {@link DocCreationResult} object containing the ID of the created document.
    */
-  public async createDoc(): Promise<DocCreationResult> {
-    const result = await this.doFetch('doc/new', { method: 'POST' })
```

Comment:
This only worked by mistake: `doFetch` sends a POST request if the body is defined, and this sets the body to `{method: POST}`. The server ignores the `method` field.

Reply:
I think I'd vote for just having `doFetch` accept a method (like `fetch()` does) rather than having to pass an empty object as the body in order to "hack around" the faulty heuristic of setting the method based on the body. Thoughts?

Reply:
Also, I think the `getClientToken()` method below is using `doFetch()` incorrectly (passing what looks like FetchOptions as the request body).

Reply:
My general principle for the API is that _every_ POST request should have a valid JSON body, even if it is an empty object. Since `doFetch` is not a general-purpose fetch function but one designed for this API, the GET/POST distinction restricts the degrees of freedom that the caller has to break things.

Reply:
> Also, I think the `getClientToken()` method below is using `doFetch()` incorrectly (passing what looks like FetchOptions as the request body).

Good point, here's a fix: https://github.com/drifting-in-space/y-sweet/pull/145

Reply:
I'm not sure it restricts degrees of freedom for the caller. After all, the caller of `doFetch()` can still break things by forgetting to pass an empty object.

My main issue is that it's just easy for the caller of `doFetch` to make bad assumptions about what `doFetch` will do. At this point in the logic, all the dev cares about are the 3 main components of a REST endpoint: method, path, and request body. Seems to me that designing a `doFetch` interface that looks like that is more likely to result in a dev correctly predicting what the code will do.

Reply:
Are you thinking that the signature should be `doFetch(url, method, body?)`? I guess I could use typescript overloads to ensure that a body is passed if and only if the method is `POST`.

Alternatively, I could use `doGet(url)` and `doPost(url, body)`, although they'd still use a shared underlying `doFetch`, so we could still shoot ourselves in the foot by calling the underlying `doFetch` instead of `doGet` or `doPost`.

Reply:
I like the update you pushed